onnxruntime/python/tools/quantization/calibrate.py

```diff
@@ -0,0 +1,232 @@
+    :param quantization_thresholds:
+        Dictionary specifying the min and max values for outputs of conv and matmul nodes.
+        The quantization_thresholds should be specified in the following format:
+            {
+                "param_name": [min, max]
+            }
+        example:
+            {
+                'Conv_3:0': [np.float32(0), np.float32(0.5)],
+                'Conv_4:0': [np.float32(1), np.float32(3.5)]
+            }
+    :return: Dictionary containing the zero point and scale values for outputs of conv and matmul nodes.
+        The dictionary format is
+            {
+                "param_name": [zero_point, scale]
+            }
+    '''
+    if nbits != 8:
+        raise ValueError('Unknown value for nbits. only 8 bit quantization is currently supported')
+
+    if quantization_thresholds == None:
+        raise ValueError('output quantization threshold is required to calculate quantization thresholds')
+
+    quantization_params = {}
+    for index, node in enumerate(model.graph.node):
+        node_output_name = node.output[0]
+        if node_output_name in quantization_thresholds:
+            node_thresholds = quantization_thresholds[node_output_name]
+            node_params = calculate_scale_zeropoint(node, model.graph.node[index+1], node_thresholds[0], node_thresholds[1])
+            quantization_params[node_output_name] = node_params
+
+    return quantization_params
+
+
+def load_pb_file(data_file_name, dataset_size, samples, channels, height, width):
+    '''
+    Load tensor data from pb files.
+    :param data_file_name: path to the pb file
+    :param dataset_size: number of image-data in the pb file for data size check
+    :param samples: number of samples 'N'
+    :param channels: number of channels in the image 'C'
+    :param height: image height for data size check 'H'
+    :param width: image width for data size check 'W'
+    :return input data for the model
+    '''
+    tensor = onnx.TensorProto()
+    inputs = np.empty(0)
+    with open(data_file_name, 'rb') as fin:
+        tensor.ParseFromString(fin.read())
+        inputs = numpy_helper.to_array(tensor)
+        try:
+            shape = inputs.shape
+            inputs = inputs.reshape(dataset_size, samples, channels, height, width)
+        except:
+            sys.exit("Input .pb file contains incorrect input size. \nThe required size is: (%s). The real size is: (%s)"
+                        %((dataset_size, samples, channels, height, width), shape))
+
+    return inputs
+
+def main():
+    # Parsing command-line arguments
+    parser = argparse.ArgumentParser(description='parsing model and test data set paths')
+    parser.add_argument('--model_path', required=True)
+    parser.add_argument('--dataset_path', required=True)
+    parser.add_argument('--output_model_path', type=str, default='calibrated_quantized_model.onnx')
+    parser.add_argument('--calib_mode', default='naive')
```

Comment:
Add a parameter to verify value for calib_mode: choices=['naive', 'smooth']

Reply:
I removed smooth option... now the script returns an error if the value is anything other than naive... I want to keep the mode here because the current mode is a very basic way of selecting min max values ... there is potential to explore other techniques to improve this... so lets keep it this way for now